Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm

```diff
@@ -158,8 +169,18 @@ static inline bool canHandleResponse(const ResourceResponse& response)
 void ParentalControlsContentFilter::finishedAddingData()
 {
 #if HAVE(WEBCONTENTRESTRICTIONS)
-    if (m_usesWebContentRestrictions)
+    if (m_usesWebContentRestrictions) {
+        if (m_state != State::Filtering)
+            return;
+
+        // Callers expect state is ready after finishing adding data.
+        Locker resultLocker { m_resultLock };
+        while (!m_isAllowdByWebContentRestrictions)
+            m_resultCondition.wait(m_resultLock);
+        m_state = *m_isAllowdByWebContentRestrictions ? State::Allowed : State::Blocked;
+        m_replacementData = std::exchange(m_webContentRestrictionsReplacementData, nullptr);
```

Comment:
In a possible follow-up patch, would it be feasible to add a completion handler to `finishedAddingData`, so that we don't have to block the main thread? This does not block landing this PR, though.

Reply:
It's possible but would require some amount of refactoring (at all callers), so this seems a safer change right now. We should consider about it later, and I will file a bug about this.
Thanks for review!